lib/prism.js

```diff
@@ -0,0 +1,89 @@
+'use strict';
+
+const Prism = require('prismjs');
+const prismLoadLanguages = require('prismjs/components/');
+
+// https://github.com/PrismJS/prism/issues/2145
+const prismComponents = require('prismjs/components');
+const prismSupportedLanguages = Object.keys(prismComponents.languages);
+
+const escapeHTML = require('./escape_html');
+const unescapeHTML = require('./unescape_html');
+
+/**
+ * Wrapper of Prism.highlight()
+ * @param {String} code
+ * @param {String} language
+ */
+function prismHighlight(code, language) {
+  console.log(Object.keys(Prism.languages));
+  // Prism has not load the language pattern
+  if (!Prism.languages[language] && prismSupportedLanguages.includes(language)) prismLoadLanguages(language);
+
+  if (Prism.languages[language]) {
+    // Prism escapes output by default
+    return Prism.highlight(unescapeHTML(code), Prism.languages[language], language);
+  }
+
+  // Current language is not supported by Prism, return origin code;
+  return escapeHTML(code);
+}
+
+/**
+ * Generate line number required HTML snippet
+ * @param {String} code - Highlighted code
+ */
+function lineNumberUtil(code) {
+  const matched = code.match(/\n(?!$)/g);
+  const num = matched ? matched.length + 1 : 1;
+  const lines = new Array(num + 1);
+
+  return `<span aria-hidden="true" class="line-numbers-rows">${lines.join('<span></span>')}</span>`;
```

Comment:
This approach (instead of table) is used to support prismjs css layout? Or it's recommended by prismjs?
Can this support https://github.com/hexojs/hexo-util/issues/145?

Reply:
> This approach (instead of table) is used to support prismjs css layout?

Specifically, `prism-line-numbers.css`: https://github.com/PrismJS/prism/blob/master/plugins/line-numbers/prism-line-numbers.css

PrismJS is using some CSS3 feature to render line number.

Reply:
In `prism-line-numbers.css` line numbers are applied with `user-select: none`.

After then, it depends on the theme maintainer.